### ISSUE TITLE
AP does not work in A319 and A321 because FBW does not

### ISSUE THREAD

**Issue by jan-hudec** [FGMEMBERS/A320neo#6]
This is weird problem, because I am not able to reproduce it consistently.

Every time it happened I had:
- Route manager is set up, default departure is used which fills in altitudes for the departure fixes, but en-route fixes _don't_ have altitude set.
- Speed pulled and left at the default 250 knots
- Heading pushed (LNAV, heading dashed out)
- Altitude left at the default 10000 and pushed (altitude still shows 10000, VS dashed out)
- AP1 engaged soon after take off

The nose kept rising until the aircraft stalled.

But other times I tries it kept climbing and accelerating as expected and I am not sure what I may have done differently.

**Comment by pedromorgan** [comment]
Can u supply more info.. 

1) what is the vertical speed
2) what "pitch up" does the aircaft kinda get to when it stalls..
3) what altitide...

My understaning is that most airliners on takeoff/climb and "srs" (speed reference system),
so after takeoff, the nose is raised or lowered to maintain speed..

**Comment by pedromorgan** [comment]
oops also mean to add.. Whilst on the runway, can you set the vs to say 2000fpm or 3000fpm and then switch..

**Comment by jan-hudec** [comment]
Vertical speed exceeded 5,500 ft/min, pitch exceeded 30°, not sure by how much as the AI was all blue. The pitch was added progressively. The plane flew for a while at ~160 knots, just above flaps-up stall speed, and the pitch slowly raised while the speed decayed until the stall warning sounded after which the AP still seemed to keep pulling up. I had to turn it off manually.

Altitude was few thousand feet. However the first time it happened I turned AP off, flew manually for a while and later tried to engage it again at ~FL160 and it again started pulling up until stall.

When I push the _altitude_ knob (which turns the altitude restrictions purple as I'd expect), the _vertical speed_ goes blank. I didn't yet see the problem with the altitude knob pulled.

**Comment by jan-hudec** [comment]
Even when it seems to work reasonably it does not seem to respect the selected vertical speed and the selected vertical speed keeps disappearing (both managed and selected altitude).

**Comment by jan-hudec** [comment]
Another problem: This time it climbed fine to FL340 and levelled off, but then it the climb rate kept oscillating between 0 and smaller positive values (+700 or so) and the aircraft kept slowly gaining altitude. When it gained ~1200 ft (FL352) a diverging oscillation in pitch started with progressively. I paused it when the V/S oscillations were about -1000 to +1500 ft/min, the aircraft still slightly climbing on average. I would write down some internal properties, but I don't know which ones.

**Comment by jan-hudec** [comment]
Ok, the oscillation is clearly problem of the elevator PID controller itself. The `/servo-control/elevator-vs` is true, `/flight-management/control/speed-with-pitch` is false, so the `FCU/AP Elevator Control VS` controller should be enabled, `/servo-control/target-vs` gradually decreased to `-50`, yet the `/velocities/vertical-speed-fps` oscillates between about -15 and +20.

Looking at the controller the `Td` appears absurdly low (elevator directly controls roll acceleration and the aircraft has a lot of inertia (and I am testing this on A321, which has highest), so some prediction is definitely needed for proper damping.

I am also not sure about the limits. If the plane was properly trimmed, the elevator could be kept within tight limits, but it is not as the FBW on `master` only trims when all `/controls/flight` inputs are 0 and they are not.

**I tried adjusting the values to `<Td>1</Td>` and slightly reducing sensitivity to `<Kp>-0.005</Kp>` and that seems to work fine, though I have to do more tests.**

However I have a slight problem with this. The `Td` of `1` basically means “estimate the input 1 second ahead using its current derivative and use that as targets”. Without working FDM it is needed, because the plane exhibits rather pronounced and rapid phugoid oscillations. However when the FBW is finished, there will be another controller in the chain to take care of that so this one shouldn't predict so much again.

And the same goes for the elevator limits. Without FBW the autopilot needs to be allowed to move the elevator in large range, because the plane may be out of trim and need the elevator input just to remain stable. But the missing FBW controller should be taking care of that, so then the values in A/P will be correct again.

Unfortunately making the FBW is _hard_. I have looked at it some time ago and didn't get far. The problem is that the A/P controller only needs to be tuned approximately, because it does not matter much if the autopilot takes a second or two to stabilize. But the FBW controller needs to react almost immediately _and_ must not cause oscillations to be controllable by hand and that can't be tuned by trial and error and probably requires speed- and weight-dependent coefficients.

**Comment by jan-hudec** [comment]
Oh, I see it now. After more testing I realized, that the FBW works in A320neo, but not in A319neo and A321neo. Therefore the autopilot, which is designed to control the FBW, also works fine on A320neo, but not A319neo and A321neo.

The first time I had problems it was with A319neo. Then I tried the A320neo a bit and it worked, but afterwards I picked A321neo and it showed the problem again which prompted me to report it.

I'll look into unifying the set files so that all three use the same, correct, flight control system (and generally eliminate any other differences that don't make sense to be there).
